content/blog/2018-02-07-strict-mode.md

```diff
@@ -0,0 +1,87 @@
+
+> Note:
+>
+> Strict mode checks are run in development mode only; _they do not impact the production build_.
+
+You can enable strict mode for any part of your application. For example:
+`embed:update-on-async-rendering/enabling-strict-mode.js`
+
+In the above example, strict mode checks will *not* be run against the `Header` and `Footer` components. However, `RouteOne` and `RouteTwo`, as well as all of their descendants, will have the checks.
+
+In version 16.3, `StrictMode` helps with:
+* [Identifying components with unsafe lifecycles](#identifying-unsafe-lifecycles)
+* [Warning about legacy string ref API usage](#warning-about-legacy-string-ref-api-usage)
+* [Detecting unexpected side effects](#detecting-unexpected-side-effects)
+
+Additional functionality will be added with future releases of React.
+
+### Identifying unsafe lifecycles
+
+As explained [in this blog post](/blog/2018/02/07/update-on-async-rendering.html), certain legacy lifecycle methods are unsafe for use in async React applications. However, if your application uses third party libraries, it can be difficult to ensure that these lifecycles aren't being used. Fortunately, strict mode can help with this!
+
+When strict mode is enabled, React compiles a list of all class components using the unsafe lifecycles, and logs a warning message with information about these components, like so:
+
+![](../images/blog/strict-mode-unsafe-lifecycles-warning.png)
+
+Addressing the issues identified by strict mode _now_ will make it easier for you to take advantage of async rendering in future releases of React.
+
+### Warning about legacy string ref API usage
+
+Previously, React provided two ways for managing refs: the legacy string ref API and the callback API. Although the string ref API was the more convenient of the two, it had [several downsides](https://github.com/facebook/react/issues/1373) and so our official recomendation was to [use the callback form instead](https://reactjs.org/docs/refs-and-the-dom.html#legacy-api-string-refs).
+
+Version 16.3 adds a new option for managing refs that offers the convenience of a string ref without any of the downsides:
+`embed:16-3-release-blog-create-ref.js`
+
+> **Note:**
+>
+> Callback refs will continue to be supported in addition to the new `createRef` API.
+>
+> You don't need to replace callback refs in your components. They are slightly more flexible, so they will remain as an advanced feature.
+
+Since object refs were largely added as a replacement for string refs, strict mode now warns about usage of string refs.
+
+[Learn more about the new `createRef` API here.](#)
+
+### Detecting unexpected side effects
+
+Conceptually, React does work in two phases:
+* The **render** phase determines what changes need to be made to e.g. the DOM. During this phase, React calls lifecycles like `getDerivedStateFromProps` and `render` and then compares the result to the previous render.
```

Comment:
Maybe but render first here? It’s the one people are familiar with. Not sure it’s even important to mention gDSFP here.

Reply:
Oops. “Put”, not “but” 

After rereading I think let’s just skip gDSFP in this sentence, IMO it’s distracting because we’re trying to explain a new concept (phases) and adding another new concept (gDSFP) makes the message harder to understand